ext/standard/basic_functions.c

```diff
@@ -1214,18 +1216,38 @@ PHP_FUNCTION(flush)
 /* {{{ Delay for a given number of seconds */
 PHP_FUNCTION(sleep)
 {
-	zend_long num;
+	double seconds;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_LONG(num)
+		Z_PARAM_DOUBLE(seconds)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (num < 0) {
+	if (!(seconds >= 0)) {
```

Comment:
Please don't negative conditions if at all possible. And definitely not for things that were already sensibly written.

Reply:
Welcome to the land of IEEE 754 where `x < 0` and `!(x >= 0)` are not actually the same thing :) This probably needs a comment that the condition is written this way to handle NAN correctly.

Reply:
done - good so?

Reply:
for better readability i changed the conditon to
```
if (isnan(seconds) || seconds < 0) {
```